### PR TITLE
Refactor result sets

### PR DESCRIPTION
**Pull request by vchebotarev**
I've refactored class dependencies in result sets and dropped useless fetch constants

### CODE REVIEW

**Comment by vchebotarev** [comment]
@oohnoitz ping

**Comment by vchebotarev** [comment]
@oohnoitz ping ping

File: src/Drivers/Mysqli/Connection.php [hunk 1]
```diff
@@ -119,7 +121,7 @@ public function multiQuery(array $queue)
                 $this->getConnection()->error.' [ '.implode(';', $queue).']');
```
Comment (oohnoitz) [inline src/Drivers/Mysqli/Connection.php]: It isn't part of this PR, but we should probably look into whether it is safe to remove this considering we no longer support HHVM.

File: src/Drivers/Pdo/Connection.php [hunk 2]
```diff
@@ -26,7 +28,7 @@ public function query($query)
             throw new DatabaseException($exception->getMessage() . ' [' . $query . ']');
         }
 
-        return new ResultSet($stm);
+        return new ResultSet(new ResultSetAdapter($stm));
```
Comment (oohnoitz) [inline src/Drivers/Pdo/Connection.php]: s/$stm/$statement/g

nit: It seems weird we are using an abbreviation here and don't in other places in this file. We should change that.

**Comment by vchebotarev** [comment]
@oohnoitz done, check it, please

**Pull request merged**